tests/jax/components/building/environments_test.py

```diff
@@ -0,0 +1,217 @@
+from mava.utils.sort_utils import sort_str_num
+
+
+class AbstractExecutorEnvironmentLoop(ExecutorEnvironmentLoop):
+    """Implement abstract methods to allow testing of class"""
+
+    def on_building_executor_environment_loop(self, builder: SystemBuilder) -> None:
+        """Do nothing. Just implement abstract method"""
+        pass
+
+
+@pytest.fixture
+def test_environment_spec() -> EnvironmentSpec:
+    """Pytest fixture for environment spec"""
+    environment_factory = functools.partial(
+        debugging_utils.make_environment,
+        env_name="simple_spread",
+        action_space="discrete",
+        num_agents=10,
+    )
+
+    config = EnvironmentSpecConfig(environment_factory=environment_factory)
+    test_environment_spec = EnvironmentSpec(config=config)
+    return test_environment_spec
+
+
+@pytest.fixture
+def test_executor_environment_loop() -> ExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+    config = ExecutorEnvironmentLoopConfig(should_update=False)
+    return AbstractExecutorEnvironmentLoop(config=config)
+
+
+@pytest.fixture
+def test_parallel_executor_environment_loop() -> ParallelExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+
+    class ExecutorStatsWrapperClass:
+        def __init__(self, executor_environment_loop: Any) -> None:
+            self.executor_environment_loop = executor_environment_loop
+            self.wrapped = True
+
+    config = ExecutorEnvironmentLoopConfig(
+        should_update=False,
+        executor_stats_wrapper_class=ExecutorStatsWrapperClass,  # type: ignore
+    )
+    return ParallelExecutorEnvironmentLoop(config=config)
+
+
+@pytest.fixture
+def test_builder() -> SystemBuilder:
+    """Pytest fixture for system builder."""
+
+    def environment_factory(evaluation: bool) -> str:
+        return "environment_eval_" + ("true" if evaluation else "false")
+
+    global_config = SimpleNamespace(environment_factory=environment_factory)
+    system_builder = Builder(components=[], global_config=global_config)
+    system_builder.store.executor_environment = "environment"
+    system_builder.store.executor = "executor"
+    system_builder.store.executor_logger = "executor_logger"
+
+    return system_builder
+
+
+class TestEnvironmentSpec:
+    """Tests for EnvironmentSpec"""
+
+    def test_init(self, test_environment_spec: EnvironmentSpec) -> None:
+        """Test that class loads config properly"""
+
+        environment = test_environment_spec.config.environment_factory()
+        assert environment.environment.num_agents == 10
+
+    def test_on_building_init_start(
+        self, test_environment_spec: EnvironmentSpec, test_builder: SystemBuilder
+    ) -> None:
+        """Test by manually calling the hook and checking the store."""
+        test_environment_spec.on_building_init_start(test_builder)
+
+        # Assert for type and extra spec
+        environment_spec = test_builder.store.environment_spec
```

Comment:
Due to the env specs renaming PR this test currently has errors. Please update the names to match the new naming :)